pkg/types/time.go

```diff
@@ -1168,7 +1183,11 @@ func parseDatetime(ctx Context, str string, fsp int, isFloat bool) (Time, error)
 	if hhmmss {
 		// If input string is "20170118.999", without hhmmss, fsp is meaningless.
 		// TODO: this case is not only meaningless, but erroneous, please confirm.
-		microsecond, overflow, err = ParseFrac(fracStr, fsp)
+		truncate := false
```

Comment:
seems this is `truncate := ctx.Flags().TimeTruncateFractional()`

Reply:
+1.